app/src/main/java/com/example/cookbook/ui/create_recipe/OwnRecipesViewModel.kt

```diff
@@ -24,22 +25,33 @@ class OwnRecipesViewModel @Inject constructor(private val recipeRepository: OwnR
     val isSuccessful: LiveData<Boolean>
         get() = _isSuccessful
 
+    private val composite = CompositeDisposable()
+
     init {
         observeRecipes()
     }
 
     private fun observeRecipes() {
         _isLoading.value = true
-        viewModelScope.launch {
-            ownRecipes = recipeRepository.getRecipeListSync()
-            _isLoading.value = false
-        }
+
+        val disposable = recipeRepository.getRecipeListSync()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe {
+                ownRecipes.postValue(it)
+                _isLoading.value = false
```

Comment:
You can use  `.doOnSubscribe {  }` and ` .doOnUnsubscribe {}` method for setting value to` _isLoading` livedata